voltdb/src/main/java/com/yahoo/ycsb/db/voltdb/ConnectionHelper.java

```diff
@@ -0,0 +1,215 @@
+   */
+  private static void connectToOneServerWithRetry(final Client client, String server) {
+    int sleep = 1000;
+    while (true) {
+      try {
+        client.createConnection(server);
+        break;
+      } catch (Exception e) {
+        System.err.printf("Connection failed - retrying in %d second(s).\n", sleep / 1000);
+        try {
+          Thread.sleep(sleep);
+        } catch (java.lang.InterruptedException e2) {
+          e.printStackTrace();
+        }
+        if (sleep < 8000) {
+          sleep += sleep;
+        }
+      }
+    }
+    System.out.printf("Connected to VoltDB node at: %s.\n", server);
+  }
+
+  /**
+   * See if DB servers are present on the network.
+   * 
+   * @return true or false
+   */
+  public static boolean checkDBServers(String servernames) {
+
+    String[] serverNamesArray = servernames.split(",");
+
+    boolean dbThere = false;
+
+    Socket socket = null;
+    try {
+      // Connect
+      socket = new Socket(serverNamesArray[0], VOLTDB_DEFAULT_PORT);
+      dbThere = true;
+    } catch (IOException connectFailed) {
+      dbThere = false;
+    } finally {
+      if (socket != null) {
+        try {
+          socket.close();
+        } catch (IOException ignore) {
+          // Ignore.
+        }
+      }
+      socket = null;
+    }
+
+    return dbThere;
+
+  }
+
+  /**
+   * Connect to a set of servers in parallel. Each will retry until connection.
+   * This call will block until all have connected.
+   *
+   * @param servers A comma separated list of servers using the hostname:port
+   *                syntax (where :port is optional).
+   * @throws InterruptedException if anything bad happens with the threads.
+   */
+  private static void connect(final Client client, String servers) throws InterruptedException {
+    System.out.println("Connecting to VoltDB...");
+
+    String[] serverArray = servers.split(",");
+    final CountDownLatch connections = new CountDownLatch(serverArray.length);
+
+    // use a new thread to connect to each server
+    for (final String server : serverArray) {
+      new Thread(new Runnable() {
+        @Override
+        public void run() {
+          connectToOneServerWithRetry(client, server);
+          connections.countDown();
+        }
+      }).start();
+    }
+    // block until all have connected
+    connections.await();
+  }
+
+  private static class ClientConnection {
+    private Client mclient;
+    private AtomicInteger mconnectionCount;
+
+    ClientConnection(Client client) {
+      mclient = client;
+      mconnectionCount = new AtomicInteger(1);
+    }
+
+    void connect() {
+      mconnectionCount.incrementAndGet();
+    }
+
+    void disconnect() {
+      int count = mconnectionCount.decrementAndGet();
+      if (count <= 0 && mclient != null) {
+        synchronized (this) {
+          try {
+            mclient.drain();
+            mclient.close();
+          } catch (NoConnectionsException e) {
+            e.printStackTrace();
```

Comment:
Please log via a logging framework instead of using printStackTrace directly

Reply:
Fixed in 0ebc1ec